src/timezones/types.jl

```diff
@@ -3,6 +3,21 @@
 using Base.Dates
 import Base.Dates: value
 
+
+abstract TimeError <: Exception
+
+type AmbiguousTimeError <: TimeError
+    dt::DateTime
+    tz::TimeZone
+end
+Base.showerror(io::IO, e::AmbiguousTimeError) = print(io, "Local DateTime $(e.dt) is ambiguious");
```

Comment:
The `DateTime`s in these exceptions is just expected to be in localtime. We may want to extend the exceptions to be aware of what kind of `DateTime`s they contain (local/UTC). Most of the time you only get into trouble using local `DateTime`s which is why that is what is expected. There are some odd interactions within `type.jl` such as:

```julia
throw(NonExistentTimeError(localtime(zdt), tz))
```